### ISSUE TITLE
undefined is not an object (evaluating 'RNRandomBytes.seed') - iOS

### ISSUE THREAD

**Issue by ayushkhare** [mvayngrib/react-native-randombytes#19]
This happens after I link it manually on iOS as specified here

http://facebook.github.io/react-native/docs/linking-libraries-ios.html#content

**Comment by mvayngrib** [comment]
@ayushkhare just to clarify, you linked manually in XCode by dragging the .a file from RNRandomBytes Products into Link Binary With Libraries?

**Comment by ayushkhare** [comment]
@mvayngrib Yes I dragged and dropped and linked it manually. I am attaching screenshots below just for clarifications


**`I have dragged and dropped the .xcodeproj inside the libraries folder of my main project`**
<img width="336" alt="screen shot 2018-02-03 at 12 08 38 pm" src="https://user-images.githubusercontent.com/9208345/35762925-b0626c26-08db-11e8-85a3-ce84283b7aa7.png">

**`I have included the .a file in Link Binary with LIbraries`**
<img width="728" alt="screen shot 2018-02-03 at 12 08 01 pm" src="https://user-images.githubusercontent.com/9208345/35762911-44356152-08db-11e8-9124-669d12a50321.png">



**`This is what my header search path looks like for my main project`**

<img width="501" alt="screen shot 2018-02-03 at 12 08 23 pm" src="https://user-images.githubusercontent.com/9208345/35762916-761abbfe-08db-11e8-82dd-4ea798d8aff4.png">


**`This is what my header search path looks like for random bytes project`**

<img width="507" alt="screen shot 2018-02-03 at 12 08 56 pm" src="https://user-images.githubusercontent.com/9208345/35762937-fc912772-08db-11e8-845b-e787a981ec9f.png">


After doing all this I get the following error

<img width="356" alt="screen shot 2018-02-03 at 12 07 46 pm" src="https://user-images.githubusercontent.com/9208345/35762943-0b96de7e-08dc-11e8-9fc7-b4cfce50b0b6.png">

**Comment by mvayngrib** [comment]
@ayushkhare i have the same header search paths in my project. What version of react-native? If your project is not private, I can take a quick look.

**Comment by ayushkhare** [comment]
@mvayngrib I am using version 0.50.4, and sorry but I cannot share the project with you! I am pretty sure something is wrong with the linking. I was able to link it successfully on my Android project, and it runs fine. Can you suggest what could be possibly wrong?

**Comment by mvayngrib** [comment]
@ayushkhare not sure. Can you try https://github.com/mvayngrib/rnctest1 and see if it works for you on iOS? Then try upgrading it to 0.50.4 (it's on RN 0.48 now), and see if it breaks.  If it does, maybe you can figure out the difference in linking. Let me know if you find something!

**Comment by ayushkhare** [comment]
@mvayngrib I couldn't really figure out what's wrong, so I just copied the `header` and `.m` file in the folder where I have my other native modules, and everything works fine. Thanks!

**Comment by mvayngrib** [comment]
@ayushkhare ok, glad you got it working one way or another :)

**Comment by brandonfancher** [comment]
I'm getting the same error. Here's the error from by build log:

![image](https://user-images.githubusercontent.com/7911424/37264814-fc815d3a-2585-11e8-9952-cf39a2840593.png)

**Comment by brandonfancher** [comment]
Believe it or not, changing all instances of `#import "RCTBridgeModule.h"` back to `#import <React/RCTBridgeModule.h>` seems to have fixed the build for me.

**Comment by you-fail-me** [comment]
Have the very same problem, RN 54 as well. What is the reason and the recommended fix?

**Comment by wym7230** [comment]
I'm getting the same error. I initiated an empty project using expo XDE. Ejected the project, ran the workflow in readme. Open Xcode to build. and get the same error.

**Comment by imansalhi** [comment]
same issue

**Comment by dotpegaso** [comment]
same issue here...

**Comment by never00miss** [comment]
@dotpegaso maybe you can use costumed random bytes on my repo, use native js without android or iOS bridge

**Comment by dotpegaso** [comment]
@never00miss I'm afraid that I can't use that.
-
There's really no solution for this case yet?

**Comment by never00miss** [comment]
@dotpegaso What you mean that you "afraid" of?
On Android nor IOS, all the native function that defined on rnrandombytes only trigger seed function that exactly the same that I am doing on nativeJS.

You can see here
https://github.com/mvayngrib/react-native-randombytes/blob/master/android/src/main/java/com/bitgo/randombytes/RandomBytesModule.java at line 39.

What was I doing same as native function.

**Comment by mvayngrib** [comment]
@dotpegaso are you on react native 0.60 by any chance? Because this lib has only been updated to work with RN 0.59.x

**Comment by never00miss** [comment]
@mvayngrib 😑, are you sure that you was see my source code on my repo? What I know that RN 0.4x until now 0.6x do a lot of improvement on their native bridge. "What I definitely define here" my fork not using any "Native bridge to work for".. so thank you.

**Comment by mvayngrib** [comment]
@never00miss sorry, i didn't understand that :(

i'm not saying RN 0.60 is worse. I'm saying it handles native modules differently, and that I haven't tested react-native-randombytes with RN 0.60

> there is no secure random in "client side"

your fork does not produce a secure random because it uses Math.random to build the seed. This fork does, because it uses the device's native rng :) There are plenty of production applications that rely on producing secure random numbers on the client side (for example, the major cryptocurrency wallets)

**Comment by never00miss** [comment]
@mvayngrib "A cryptographically secure number random generator, as you might use for generating encryption keys, works by gathering entropy - that is, unpredictable input - from a source which other people can't observe."

There is no guarantee that the random number generator cannot be influenced by other scripts, that's mean device "rooted" or other things that cut of V8 engines kernel,  Math random using A property of PRNGs is that they have a fixed internal cycle length which means once you generate a certain amount of "randomness", you go back and generate the same thing again. In order to be cryptographically secure, a random number must not leak information about what the next random number will be. Since this is not the case for a PRNG, a different source of randomness is needed (such as gathering entropy from hardware timings or other unpredictable things or hardware that measures physical properties that behave randomly). And

I still improving that using a '''window.crypto.getRandomValues''' but the problem I need to rn-nodify that crypto for using it on RN, so problem case still not solve yet, so we create a randomness on "server-side", that's what I mean on comment before

**Comment by mvayngrib** [comment]
> There is no guarantee that the random number generator cannot be influenced by other scripts, that's mean device "rooted" or other things that cut of V8 engines kernel

the goal of this library is to provide access to the native RNG on iOS/Android. Beyond that, it doesn't make any claims or guarantees

**Comment by never00miss** [comment]
@mvayngrib ok then, goodluck
